doc/reference.md

```diff
@@ -60,6 +62,37 @@ elaboration, and core language is forthcoming.
 - [References](#references)
 - [Void](#void)
 
+## Modules
+
+Fathom modules are made up of multiple top-level items
+
+### Definitions
+
+Top-level definitions are preceded by the `def` keyword:
+
+```fathom
+def point = {
+  x <- u32be,
+  y <- u32be,
+};
+```
+
+Definitions can optionally have a type annotation given to them:
+
+```fathom
+def point : Format = {
+  x <- u32be,
+  y <- u32be,
+};
+```
+
+Definitions with `_` in place of the name are still type checked, but cannot be
+used in other parts of the Fathom module:
+
+```fathom
+def _ = u32be;
```

Comment:
Oh.... um... I can't remember why I added this… must have been in a strange mood at the time. It's a common  thing to have in other langs like Agda for writing examples and test cases, but I might just remove it for now.

Reply:
I think maybe it was because at the time I was trying to convert the integration tests to use modules, and some of them used `_` for some throwaway test defs, and it seemed easy enough to add. But yeah might be a bit frivolous atm.